src/lib/integration_test_cloud_engine/kubernetes_network.ml

```diff
@@ -1,15 +1,246 @@
+open Async
+open Core
+
 module Node = struct
+  (* a pod name, e.g., test-block-producer-1 *)
   type t = string
 
+  let to_string t = t
+
+  let full_name_opt : string option ref = ref None
```

Comment:
Is this a global ref to cache the first name that you attempt to access? Not only does it not make sense to cache a full pod id (because it can change underneath you in kubernetes), but this will not work as soon as we try and send payments to 2 different nodes in the same test. Even if you did want to cache this (which I think we should avoid, so we don't need to deal with invalidation), you should cache it on the `Node.t` type instead of like this.